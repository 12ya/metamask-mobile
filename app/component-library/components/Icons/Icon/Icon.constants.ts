/* eslint-disable import/prefer-default-export */
// Internal dependencies.
import { IconName, IconProps } from './Icon.types';

export const TEST_ICON_PROPS: IconProps = {
  name: IconName.AddOutline,
};
