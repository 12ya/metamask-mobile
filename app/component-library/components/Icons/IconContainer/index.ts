export { default } from './IconContainer';
export { IconContainerSizes } from './IconContainer.types';
