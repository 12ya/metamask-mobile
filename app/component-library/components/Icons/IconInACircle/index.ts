export { default } from './IconInACircle';
export { IconInACircleSizes } from './IconInACircle.types';
