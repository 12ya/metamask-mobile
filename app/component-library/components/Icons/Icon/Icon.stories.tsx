// Third party dependencies.
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { color, select } from '@storybook/addon-knobs';

// External dependencies.
import { storybookPropsGroupID } from '../../../constants/storybook.constants';

// Internal dependencies.
import Icon from './Icon';
import { IconProps, IconSize, IconName } from './Icon.types';

export const getIconStoryProps = (): IconProps => ({
  size: select('size', IconSize, IconSize.Md, storybookPropsGroupID),
  name: select('name', IconName, IconName.LockFilled, storybookPropsGroupID),
  color: color('color', '', storybookPropsGroupID),
});
const IconStory = () => <Icon {...getIconStoryProps()} />;

storiesOf('Component Library / Icons', module).add('Icon', IconStory);

export default IconStory;
