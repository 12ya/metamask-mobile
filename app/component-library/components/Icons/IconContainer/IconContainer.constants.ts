/* eslint-disable import/prefer-default-export */
// External dependencies.
import { IconSize } from '../../Icons/Icon';
import { TEST_ICON_PROPS } from '../Icon/Icon.constants';

// Internal dependencies.
import { IconContainerSizes, IconContainerProps } from './IconContainer.types';

/**
 * Mapping of IconSize by IconContainerSizes.
 */
export type IconSizeByIconContainerSizes = {
  [key in IconContainerSizes]: IconSize;
};

export const ICON_SIZE_BY_ICON_CONTAINER_SIZE: IconSizeByIconContainerSizes = {
  [IconContainerSizes.Xs]: IconSize.Xs,
  [IconContainerSizes.Sm]: IconSize.Sm,
  [IconContainerSizes.Md]: IconSize.Md,
  [IconContainerSizes.Lg]: IconSize.Lg,
  [IconContainerSizes.Xl]: IconSize.Xl,
};

export const DEFAULT_ICON_CONTAINER_SIZE = IconContainerSizes.Md;

export const ICON_CONTAINER_TEST_ID = 'icon-container';
export const ICON_CONTAINER_ICON_TEST_ID = 'icon-container-icon';

export const TEST_ICON_CONTAINER_PROPS: IconContainerProps = {
  iconProps: TEST_ICON_PROPS,
};
