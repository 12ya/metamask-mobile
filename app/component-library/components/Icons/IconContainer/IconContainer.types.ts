// Third party dependencies.
import { ColorValue } from 'react-native';

// External dependencies.
import { CoinPatternProps } from '../../../patterns/Coins/Coin';
import { IconProps } from '../../Icons/Icon';

/**
 * IconContainer sizes.
 */
export { CoinPatternSizes as IconContainerSizes } from '../../../patterns/Coins/Coin/Coin.types';

/**
 * IconContainer component props.
 */
export type IconContainerProps = CoinPatternProps &
  Omit<IconProps, 'size'> & {
    /**
     * Optional enum to add color to the background of the Avatar.
     * @default theme.colors.background.alternative
     */
    backgroundColor?: ColorValue;
  };

/**
 * Style sheet input parameters.
 */
export type IconContainerStyleSheetVars = Pick<
  IconContainerProps,
  'style' | 'backgroundColor'
>;
