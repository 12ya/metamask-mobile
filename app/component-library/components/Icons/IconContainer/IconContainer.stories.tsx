// Third party dependencies.
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { select } from '@storybook/addon-knobs';

// External dependencies.
import { storybookPropsGroupID } from '../../../constants/storybook.constants';
import { IconName } from '../../Icons/Icon';

// Internal dependencies.
import IconContainer from './IconContainer';
import { IconContainerProps, IconContainerSizes } from './IconContainer.types';

export const getIconContainerStoryProps = (): IconContainerProps => {
  const sizeSelector = select(
    'size',
    IconContainerSizes,
    IconContainerSizes.Md,
    storybookPropsGroupID,
  );
  const iconNameSelector = select(
    'name',
    IconName,
    IconName.LockFilled,
    storybookPropsGroupID,
  );

  return {
    size: sizeSelector,
    name: iconNameSelector,
  };
};
const IconContainerStory = () => (
  <IconContainer {...getIconContainerStoryProps()} />
);

storiesOf('Component Library / Icons', module).add(
  'IconContainer',
  IconContainerStory,
);

export default IconContainerStory;
