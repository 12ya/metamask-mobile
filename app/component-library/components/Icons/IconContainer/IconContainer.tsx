// Third party dependencies.
import React from 'react';

// External dependencies.
import { useStyles } from '../../../hooks';
import CoinPattern from '../../../patterns/Coins/Coin/Coin';
import Icon from '../../Icons/Icon';

// Internal dependencies.
import { IconContainerProps } from './IconContainer.types';
import {
  DEFAULT_ICON_CONTAINER_SIZE,
  ICON_SIZE_BY_ICON_CONTAINER_SIZE,
  ICON_CONTAINER_TEST_ID,
  ICON_CONTAINER_ICON_TEST_ID,
} from './IconContainer.constants';
import styleSheet from './IconContainer.styles';

const IconContainer = ({
  size = DEFAULT_ICON_CONTAINER_SIZE,
  style,
  backgroundColor,
  ...restIconProps
}: IconContainerProps) => {
  const { styles } = useStyles(styleSheet, {
    style,
    backgroundColor,
  });
  return (
    <CoinPattern
      style={styles.base}
      size={size}
      testID={ICON_CONTAINER_TEST_ID}
    >
      <Icon
        testID={ICON_CONTAINER_ICON_TEST_ID}
        size={ICON_SIZE_BY_ICON_CONTAINER_SIZE[size]}
        {...restIconProps}
      />
    </CoinPattern>
  );
};

export default IconContainer;
