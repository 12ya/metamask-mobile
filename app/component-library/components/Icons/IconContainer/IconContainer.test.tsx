// Third party dependencies.
import React from 'react';
import { shallow } from 'enzyme';

// External dependencies.
import { IconName } from '../../Icons/Icon';
import { TEST_ICON_PROPS } from '../Icon/Icon.constants';

// Internal dependencies.
import IconContainer from './IconContainer';
import {
  ICON_CONTAINER_TEST_ID,
  ICON_CONTAINER_ICON_TEST_ID,
} from './IconContainer.constants';
import { IconContainerSizes } from './IconContainer.types';

describe('IconContainer - Snapshot', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <IconContainer
        size={IconContainerSizes.Md}
        iconProps={TEST_ICON_PROPS}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('IconContainer', () => {
  it('should render IconContainer component', () => {
    const wrapper = shallow(
      <IconContainer
        size={IconContainerSizes.Md}
        iconProps={TEST_ICON_PROPS}
      />,
    );
    const IconContainerComponent = wrapper.findWhere(
      (node) => node.prop('testID') === ICON_CONTAINER_TEST_ID,
    );
    expect(IconContainerComponent.exists()).toBe(true);
  });
  it('should render IconContainer with the right IconName', () => {
    const iconName = TEST_ICON_PROPS.name;
    const wrapper = shallow(
      <IconContainer
        size={IconContainerSizes.Md}
        iconProps={TEST_ICON_PROPS}
      />,
    );

    const IconContainerIconComponent = wrapper.findWhere(
      (node) => node.prop('testID') === ICON_CONTAINER_ICON_TEST_ID,
    );
    expect(IconContainerIconComponent.props().name).toBe(iconName);
  });
});
