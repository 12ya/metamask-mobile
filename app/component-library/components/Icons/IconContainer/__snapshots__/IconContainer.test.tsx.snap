// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconContainer - Snapshot should render correctly 1`] = `
<CirclePattern
  size="32"
  style={
    Object {
      "backgroundColor": "#F2F4F6",
    }
  }
  testID="icon-container"
>
  <Icon
    name="AddOutline"
    size="20"
    testID="icon-container-icon"
  />
</CirclePattern>
`;
